base_layer/p2p/src/lib.rs

```diff
@@ -57,3 +57,44 @@ pub const MAJOR_NETWORK_VERSION: u8 = 0;
 /// Minor network version. This should change with each time the network protocol has changed in a backward-compatible
 /// way.
 pub const MINOR_NETWORK_VERSION: u8 = 0;
+
+// This function returns the network wire byte for any chosen network. Increase these numbers for any given network when
+// network traffic separation is required.
+// Note: Do not re-use previous values.
+fn get_network_wire_byte(network: Network) -> Result<u8, anyhow::Error> {
+    let network_wire_byte = match network {
+        Network::MainNet => 0,
+        Network::StageNet => 40,
+        Network::NextNet => 80,
+        Network::LocalNet => 120,
+        Network::Igor => 160,
+        Network::Esmeralda => 200,
+    };
+    verify_network_wire_byte_range(network_wire_byte, network)?;
+    Ok(network_wire_byte)
+}
+
+// This function bins the range of u8 numbers for any chosen network to a valid network_wire_byte_range.
+// Note: Do not change these ranges.
+fn verify_network_wire_byte_range(network_wire_byte: u8, network: Network) -> Result<(), anyhow::Error> {
+    if network_wire_byte == 0x46 {
+        return Err(anyhow::anyhow!("Invalid network wire byte, cannot be 0x46 (E)"));
+    }
+
+    let valid = match network {
+        Network::MainNet => (0..40).contains(&network_wire_byte),
+        Network::StageNet => (40..80).contains(&network_wire_byte),
+        Network::NextNet => (80..120).contains(&network_wire_byte),
+        Network::LocalNet => (120..160).contains(&network_wire_byte),
+        Network::Igor => (160..200).contains(&network_wire_byte),
+        Network::Esmeralda => (200..240).contains(&network_wire_byte),
+    };
+    if !valid {
+        return Err(anyhow::anyhow!(
+            "Invalid network wire byte `{}` for network `{}`",
+            network_wire_byte,
+            network
+        ));
+    }
+    Ok(())
+}
```

Comment:
This should be a unit test rather - it's not necessary to check hard coded values at runtime - you can then make get_network_wire_byte infallible.